### PR TITLE
return error on propagation failure, closes #20

### LINKED ISSUE #20: Manager provisionig result on create/update and set feedback accordingly
If there is at least one propagation failure return an error instead of success.

### PR DESCRIPTION
**Pull request by andrea-patricelli**

### CODE REVIEW

File: app/src/main/resources/org/apache/syncope/client/enduser/pages/SelfResult.html [hunk 1]
```diff
@@ -17,50 +17,58 @@
...
+                        </wicket:fragment>
+                        <wicket:fragment wicket:id="errorIcon">
+                            <svg class="checkmark_error" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 52 52">
+                                <circle class="checkmark_circle_error" cx="26" cy="26" r="25" fill="none"/>
+                                <path class="checkmark_check_error" fill="none" d="M16 16 36 36 M36 16 16 36"/>
+                            </svg>
+                        </wicket:fragment>
+                        <span wicket:id="errorListContainer"></span>
+                        <wicket:fragment wicket:id="fullErrorList">
+                            <span>ERRROREEEEEEEEEEEEEEEEEEEEEE</span>
```
Comment (mdisabatino) [inline app/src/main/resources/org/apache/syncope/client/enduser/pages/SelfResult.html:61]: Possiamo correggere questa cosa?

File: app/src/main/java/org/apache/syncope/client/enduser/pages/EditChangePassword.java [hunk 2]
```diff
@@ -47,13 +52,25 @@ protected void doPwdSubmit(final AjaxRequestTarget target, final AjaxPasswordFie
             UserPatch userPatch = new UserPatch();
             userPatch.setKey(getPwdLoggedUser().getKey());
             userPatch.setPassword(passwordPatch);
-            userSelfRestClient.update(userTO.getETagValue(), userPatch);
+            ProvisioningResult<UserTO> result = userSelfRestClient.update(userTO.getETagValue(), userPatch);
+
```
[the remark this answers is not in the document]
Comment (andrea-patricelli) [reply app/src/main/java/org/apache/syncope/client/enduser/pages/EditChangePassword.java:56]: Yes, I agree. a Lot of this code could be refactored into single method, but is not so trivial to do, due to `getString()` methods for example. I'll do the best refactoring without breaking anything.

[the remark this answers is not in the document]
Comment (mdisabatino) [reply app/src/main/java/org/apache/syncope/client/enduser/pages/EditChangePassword.java:56]: `
public class PageParametersUtils {

    public static PageParameters managePageParams(
            final Component component,
            final List<PropagationStatus> failingPropagations) {
        PageParameters parameters = new PageParameters();
        parameters.add(Constants.STATUS,
                failingPropagations.isEmpty()
                        ? Constants.OPERATION_SUCCEEDED
                        : Constants.OPERATION_ERROR);
        parameters.add(Constants.NOTIFICATION_TITLE_PARAM,
                failingPropagations.isEmpty()
                        ? component.getString("self.pwd.change.success.msg")
                        : component.getString("self.pwd.change.error.msg"));
        parameters.add(Constants.NOTIFICATION_MSG_PARAM,
                failingPropagations.isEmpty()
                        ? component.getString("self.pwd.change.success")
                        : component.getString("self.pwd.change.error"));
        parameters.add(Constants.LANDING_PAGE,
                SyncopeEnduserApplication.get().getPageClass("profile", Dashboard.class).getName());
        return parameters;
    }
    `
  and
`  
              PageParameters parameters = PageParametersUtils.managePageParams(this, failingPropagations);`
  `
  could it work?

[the remark this answers is not in the document]
Comment (andrea-patricelli) [reply app/src/main/java/org/apache/syncope/client/enduser/pages/EditChangePassword.java:54]: yes, sure!

[the remark this answers is not in the document]
Comment (andrea-patricelli) [reply app/src/main/java/org/apache/syncope/client/enduser/pages/EditChangePassword.java:54]: very nice solution, going to integrate with latest refactoring

**Review by mdisabatino — APPROVED** [review]

**Pull request merged**